src/components/forms/forms.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  getFormsDataThunk,
+  deleteFormSchemaThunk,
+  saveFormDataThunk
+} from './../../redux/forms-Reducer';
+import style from './forms.module.css';
+
+function Forms(props) {
+  const [state, setState] = useState({});
+
+  if (!props.checked) {
+    props.getFormsDataThunk();
+  } else if (!state.forms) setState({ forms: [...props.forms] });
+
+  const onSubmit = (e, key) => {
+    e.preventDefault();
+    let data = {};
+    const stateCopy = { ...state };
+    stateCopy.forms[key].inputs.forEach((item, index) => {
+      if (item.value)
+        data[(item.label + key + index).toLowerCase()] = item.value;
+      item.value = '';
+    });
+    props.saveFormDataThunk(data);
+    setState({ ...stateCopy });
+  };
+  const onHandelChnage = (e, key, index) => {
+    e.preventDefault();
+    const { value } = e.target;
+    const forms = state.forms;
+    forms[key].inputs[index].value = value;
+    setState({ forms });
+  };
+  const formsRender =
+    state.forms &&
+    state.forms.map((item, key) => {
+      return (
+        <div key={key} className={style.form_box}>
+          <div className={style.form_header}>
+            <button
+              style={{ backgroundColor: 'red' }}
+              onClick={() => {
+                props.deleteFormSchemaThunk(item._id);
+      
+              }}>
+              Delete
+            </button>
+            <span className={style.form_name}>{item.formName}</span>
+            <span className={style.form_desc}>{item.formDesc}</span>
+          </div>
+          <form
+            onSubmit={e => {
+              onSubmit(e, key);
+            }}
+            className={style.form}>
+            {item.inputs.map((item, index) => {
+              return (
+                <div key={index} className={style.form_input_box}>
+                  <label htmlFor={item.label + key + index}>{item.label}</label>
+                  {(item.type === 'text' ||
+                    item.type === 'email' ||
+                    item.type === 'date') && (
```

Comment:
😄

Reply:
instead of type checking, try to always avoid multiple conditions, nested ifs, if elses, etc
here is a little trick for your case:
you can build a map of fields. by type, and then fetch the one you need.

```
const fieldComponents = {
  text: <Input />,
  email: <Textarea />
  date: Date,
  textarea: Textarea,
  select: Select,
  etc
};

// and then you can:
const FieldComponent = fieldComponents[item.type];

if (!FieldComponent) {
 throw Error(`No component found for ${type} type`);
}
```

Reply:
then you can just reneder your component below:
```
<FieldComponent {..allYourProps} />
```